services/stripe_test.go

```diff
@@ -0,0 +1,240 @@
+package services
+
+import (
+	"bytes"
+	"testing"
+
+	"boilerplate-api/internal/config"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stripe/stripe-go/v76"
+	"github.com/stripe/stripe-go/v76/form"
+)
+
+type stripeBackendMock struct {
+	mock.Mock
+}
+
+func (s *stripeBackendMock) Call(
+	method,
+	path,
+	key string,
+	params stripe.ParamsContainer,
+	v stripe.LastResponseSetter,
+) error {
+	args := s.Called(method, path, key, params, v)
+	return args.Error(0)
+}
+
+func (s *stripeBackendMock) CallStreaming(
+	method,
+	path,
+	key string,
+	params stripe.ParamsContainer,
+	v stripe.StreamingLastResponseSetter,
+) error {
+	args := s.Called(method, path, key, params, v)
+	return args.Error(0)
+}
+
+// MockEnv is a mock implementation of config.Env
+type MockEnv struct {
+	mock.Mock
+}
+
+func (m *MockEnv) GetEnvironment() string {
+	args := m.Called()
+	return args.String(0)
+}
+
+func (s *stripeBackendMock) CallRaw(
+	method,
+	path,
+	key string,
+	body *form.Values,
+	params *stripe.Params,
+	v stripe.LastResponseSetter,
+) error {
+	args := s.Called(method, path, key, params, v)
+	return args.Error(0)
+}
+
+func (s *stripeBackendMock) CallMultipart(
+	method,
+	path,
+	key,
+	boundary string,
+	body *bytes.Buffer,
+	params *stripe.Params,
+	v stripe.LastResponseSetter,
+) error {
+	args := s.Called(method, path, key, boundary, body, params, v)
+	return args.Error(0)
+}
+
+func (s *stripeBackendMock) SetMaxNetworkRetries(maxNetworkRetries int64) {
+	s.Called(maxNetworkRetries)
+}
+
+func TestCreateCustomer(t *testing.T) {
+	stripeBackendMock := new(stripeBackendMock)
+	//stripeTestBackends := &stripe.Backends{
+	//	API:     stripeBackendMock,
+	//	Connect: stripeBackendMock,
+	//	Uploads: stripeBackendMock,
+	//}
+
+	//stripeClient := client.New("sk_test", stripeTestBackends)
+
+	env := config.NewEnv(
+		".env",
+	)
+	stripeService := NewStripeService(
+		env,
+		config.GetLogger(env),
```

Comment:
config.GetLogger expects no arguments. Getting err `too many arguments in call to config.GetLogger
	have ("boilerplate-api/internal/config".Env)
	want ()`